lib/Controller/OAuthApiController.php

```diff
@@ -156,21 +156,27 @@ public function generateToken($grant_type, $code = null,
 
 				break;
 			case 'refresh_token':
+				if (preg_match('/\bmirall\b.+2\.4\.2/i', $this->request->getHeader('User-Agent'))) {
+					$statusCode = Http::STATUS_OK;
+				}
+				else {
+					$statusCode = Http::STATUS_BAD_REQUEST;
```

Comment:
How about moving this before the if condition? 
```
$statusCode = Http::STATUS_BAD_REQUEST;
if (preg_match('/\bmirall\b.+2\.4\.2/i', $this->request->getHeader('User-Agent'))) {
	$statusCode = Http::STATUS_OK;
}
```

This eases the flow of reading - and you could add a comment on top outlining that this is a fix

Reply:
Good point!

Reply:
can you tell my, how to set a http user agent header in the unit test case?